core/server/worker/src/main/java/alluxio/worker/block/annotator/LRUAnnotator.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block.annotator;
+
+import alluxio.collections.Pair;
+
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * {@link BlockAnnotator} implementation of LRU scheme.
+ */
+public class LRUAnnotator implements BlockAnnotator<LRUAnnotator.LRUSortedField> {
+  private static final Logger LOG = LoggerFactory.getLogger(LRUAnnotator.class);
+
+  private AtomicLong mLRUClock;
+
+  /**
+   * Creates a new LRU annotator.
+   */
+  public LRUAnnotator() {
+    mLRUClock = new AtomicLong(0);
+  }
+
+  @Override
+  public BlockSortedField updateSortedField(long blockId, LRUSortedField oldValue) {
+    long clockValue = 0;
+    if (oldValue != null) {
```

Comment:
When will there be no `oldValue`, does it make sense to set the clock value to 0 (most recent)?

Reply:
There will be no `oldValue` when the block is seen the first time. Recency is by having greater values, so 0 means the least recent. 

Assumption here is that a block will first be committed, and have its value annotated as 0. Value will be bumped only when it's accessed after commit.

Reply:
Is this typical for LRU algorithms, or should the block have highest recency on commit?

Reply:
This was the default in LRUEvictor.
I believe this could potentially be a configuration property.

Reply:
@ggezer For the LRU evictor:
```
@Override
  public void onCommitBlock(long sessionId, long blockId, BlockStoreLocation location) {
    // Since the temp block has been committed, update Evictor about the new added blocks
    mLRUCache.put(blockId, UNUSED_MAP_VALUE);
  }
```
Doesn't this mean newly created blocks are added to the tail of the LRU cache (most recently used)?